crates/solvers/src/domain/liquidity/mod.rs

```diff
@@ -0,0 +1,27 @@
+//! Modelling on-chain liquidity.
+
+pub mod constantproduct;
+pub mod weighted;
+
+use crate::domain::eth;
+use ethereum_types::H160;
+
+/// A source of liquidity which can be used by the solver.
+#[derive(Clone, Debug)]
+pub struct Liquidity {
+    pub id: Id,
+    pub address: H160,
+    /// Estimation of gas needed to use this liquidity on-chain.
+    pub gas: eth::Gas,
+    pub state: State,
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Id(pub String);
+
+/// The liquidity state, specific to the type of liquidity.
+#[derive(Clone, Debug)]
+pub enum State {
+    ConstantProduct(constantproduct::Pool),
+    WeightedProduct(weighted::Pool),
+}
```

Comment:
I didn't want it to be called pool, because not all on-chain liquidity we receive from the driver is a liquidity pool - namely 0x liquidity orders.

Since I plan on adding 0x limit orders to this `enum`, I came up with this name. `State` is kind of general, but I couldn't think of anything better.

Reply:
I'm leaving it like this for now, I will re-visit this soon when adding the rest of the liquidity for the next solver I want to support.